cli/internal/cmd/apply_test.go

```diff
@@ -167,5 +177,215 @@ func TestSkipPhases(t *testing.T) {
+			},
+		},
+		"qemu: all files exist": {
+			createConfig:       defaultConfig(cloudprovider.QEMU),
+			createState:        defaultState,
+			createMasterSecret: defaultMasterSecret,
+			createAdminConfig:  defaultAdminConfig,
+			createTfState:      defaultTfState,
+			flags:              applyFlags{},
+			wantPhases: skipPhases{
+				skipInitPhase:  struct{}{},
+				skipImagePhase: struct{}{}, // No image upgrades on QEMU
+			},
+		},
+		"no config file": {
+			createConfig:       func(require *require.Assertions, fh file.Handler) {},
+			createState:        defaultState,
+			createMasterSecret: defaultMasterSecret,
+			createAdminConfig:  defaultAdminConfig,
+			createTfState:      defaultTfState,
+			flags:              applyFlags{},
+			wantErr:            true,
+		},
+		"no admin config file, but mastersecret file exists": {
+			createConfig:       defaultConfig(cloudprovider.GCP),
+			createState:        defaultState,
+			createMasterSecret: defaultMasterSecret,
+			createAdminConfig:  func(require *require.Assertions, fh file.Handler) {},
+			createTfState:      defaultTfState,
+			flags:              applyFlags{},
+			wantErr:            true,
+		},
+		"no admin config file, no master secret": {
+			createConfig:       defaultConfig(cloudprovider.GCP),
+			createState:        defaultState,
+			createMasterSecret: func(require *require.Assertions, fh file.Handler) {},
+			createAdminConfig:  func(require *require.Assertions, fh file.Handler) {},
+			createTfState:      defaultTfState,
+			flags:              applyFlags{},
+			wantPhases: skipPhases{
+				skipImagePhase: struct{}{},
+				skipK8sPhase:   struct{}{},
+			},
+		},
+		"no tf state, but admin config exists": {
+			createConfig:       defaultConfig(cloudprovider.GCP),
+			createState:        defaultState,
+			createMasterSecret: defaultMasterSecret,
+			createAdminConfig:  defaultAdminConfig,
+			createTfState:      func(require *require.Assertions, fh file.Handler) {},
+			flags:              applyFlags{},
+			wantErr:            true,
+		},
+		"only config file": {
+			createConfig:       defaultConfig(cloudprovider.GCP),
+			createState:        func(require *require.Assertions, fh file.Handler) {},
+			createMasterSecret: func(require *require.Assertions, fh file.Handler) {},
+			createAdminConfig:  func(require *require.Assertions, fh file.Handler) {},
+			createTfState:      func(require *require.Assertions, fh file.Handler) {},
+			flags:              applyFlags{},
+			wantPhases: skipPhases{
+				skipImagePhase: struct{}{},
+				skipK8sPhase:   struct{}{},
+			},
+		},
+		"skip terraform": {
+			createConfig:       defaultConfig(cloudprovider.GCP),
+			createState:        defaultState,
+			createMasterSecret: func(require *require.Assertions, fh file.Handler) {},
+			createAdminConfig:  func(require *require.Assertions, fh file.Handler) {},
+			createTfState:      func(require *require.Assertions, fh file.Handler) {},
+			flags: applyFlags{
+				skipPhases: skipPhases{
```

Comment:
nit: might be good to have a constructor for this to hide the cumbersome init syntax